### PR TITLE
[Dialog] Add example for async form submissions

### PR DESCRIPTION
**Pull request by kylemh**
Reference for why I worked on this: https://github.com/radix-ui/primitives/discussions/1234#discussioncomment-3281594

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code.
- [ ] Include the URL where we can test the change in the body of your PR.
^ is this necessary ?

This pull request:

- [ ] Fixes a bug
- [ ] Adds additional features/functionality
- [X] Updates documentation or example code
- [ ] Other

### CODE REVIEW

File: data/primitives/components/dialog/1.0.0.mdx [hunk 1]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
 
 ## Examples
 
+### Asynchronous form submissions within Dialog
```
Comment (andy-hook) [inline data/primitives/components/dialog/1.0.0.mdx:373]:
```suggestion
### Close after asynchronous form submission

Use the controlled props to programatically close the Dialog after an async operation has resolved.
```

File: data/primitives/components/dialog/1.0.0.mdx [hunk 2]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
...
+
+```jsx
+import { useState } from 'react';
+import { styled } from '@stitches/react';
+import * as Dialog from '@radix-ui/react-dialog';
+
+const somePromise = async () =>
```
Comment (andy-hook) [inline data/primitives/components/dialog/1.0.0.mdx:380]:
```suggestion
const mockPromise = async () =>
```
nit

File: data/primitives/components/dialog/1.0.0.mdx [hunk 3]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
...
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+
+export default () => {
+  const [isFormDialogOpen, setFormDialogOpenState] = useState(false);
```
Comment (andy-hook) [inline data/primitives/components/dialog/1.0.0.mdx:384]:
```suggestion
  const [open, setOpen] = useState(false);
```
nit: name consistency with other demos

File: data/primitives/components/dialog/1.0.0.mdx [hunk 4]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
...
+
+  return (
+    <Dialog.Root onOpenChange={setFormDialogOpenState} open={isFormDialogOpen}>
```
Comment (andy-hook) [inline data/primitives/components/dialog/1.0.0.mdx:387]:
```suggestion
    <Dialog.Root open={open} onOpenChange={setOpen}>
```
nit (sorry): we typically to place the setter after

**Review by andy-hook — COMMENTED** [review]
Thanks for this! just a couple of small nits for consistency with other demos / code.

**Comment by kylemh** [comment]
Think I've got it all sorted for ya.

File: data/primitives/components/dialog/1.0.0.mdx [hunk 5]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
...
+```jsx
+import { useState } from 'react';
+import { styled } from '@stitches/react';
+import * as Dialog from '@radix-ui/react-dialog';
+
+const mockPromise = async () =>
+  await new Promise((resolve) => setTimeout(resolve, 1000));
+
+export default () => {
+  const [open, setOpen] = useState(false);
```
Comment (andy-hook) [inline data/primitives/components/dialog/1.0.0.mdx:384]:
```suggestion
  const [open, setOpen] = React.useState(false);
```

File: data/primitives/components/dialog/1.0.0.mdx [hunk 6]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
 
 ## Examples
 
+### Close after asynchronous form submission
+
+```jsx
+import { useState } from 'react';
```
Comment (andy-hook) [inline data/primitives/components/dialog/1.0.0.mdx:376]:
```suggestion
import React from 'react';
```

**Review by andy-hook — COMMENTED** [review]
Just a couple of others, thanks! 🙏

File: data/primitives/components/dialog/1.0.0.mdx [hunk 7]
```diff
@@ -370,6 +370,41 @@ If you want to hide the description, wrap it inside our [Visually Hidden](../uti
...
+
+  return (
+    <Dialog.Root open={open} onOpenChange={setOpen}>
+      <Dialog.Trigger />
+      <Dialog.Portal>
+        <Dialog.Overlay>
+          <Dialog.Content>
+            <form
+              onSubmit={async () => {
+                mockPromise().then(() => setFormDialogOpenState(false));
```
[the remark this answers is not in the document]
Comment (kylemh) [reply data/primitives/components/dialog/1.0.0.mdx:396]: 🤦🏼 whoops

**Review by andy-hook — APPROVED** [review]

**Pull request merged**